src/target/cortexar.c

```diff
@@ -1758,8 +1761,19 @@ static size_t cortexar_build_target_description(char *const buffer, size_t max_l
 		const char *const name = cortexr_spr_names[i];
 		const gdb_reg_type_e type = cortexr_spr_types[i];
 
-		offset += snprintf(buffer + offset, print_size, "<reg name=\"%s\" bitsize=\"32\"%s%s/>", name,
-			gdb_reg_type_strings[type], i == 3U ? " regnum=\"25\"" : "");
+		const uint8_t regnum_length = 16;
+		char *const tag_closure = malloc(regnum_length);
+
+		if (i == 3U)
+			/* Remap CPSR in particular */
+			snprintf(tag_closure, regnum_length, " regnum=\"%u\"/>", CORTEXAR_CPSR_GDB_REMAP_POS);
+		else
+			strncpy(tag_closure, "/>", regnum_length);
+
+		offset += snprintf(buffer + offset, print_size, "<reg name=\"%s\" bitsize=\"32\"%s%s", name,
+			gdb_reg_type_strings[type], tag_closure);
+
+		free(tag_closure);
```

Comment:
This change seems to produce the same result as before, just via an extra memory allocation?

Think it might be better to add a comment in front of the original snprintf() in that case to have it say what the `regnum="25"` stuff is about so it's not just a random condition in the code.

Reply:
Okay, fair. I was just aiming to reduce the number of places in which the 25 remap is defined. Will restore.

Reply:
Amended the commit.